### PR TITLE
commit files

### PR DESCRIPTION
**Pull request by fadwa2043**

### CODE REVIEW

**Review by omarmhaimdat — COMMENTED** [review]
The solution provided answers to some of the questions asked, but it is not complete. I have a couple of comments, remarks, and questions:

- I would suggest using the same naming convention for all your variables, functions, and methods. For example, you use camelCase for some of them, and snake_case for others. I would suggest using snake_case for all of them.
- Too many dependencies
- You are not leveraging GPU acceleration
- Hard coding paths are not a good practice
- Lack of post-processing (e.g. empty strings, special characters, etc.)
- Huge amount of code duplication (e.g. evaluate_ocr_model_NonArabic and evaluate_ocr_model_Arabic)
- String concatenation rather than using f-strings (interpolation)
- You didn't follow all the instructions in the README file:
    - You have renamed `input` folder to `inputs`
    - `output.json` should contain all the results for all the images

I have many more comments, but I will stop here. I hope this helps.

In conclusion, I think there is a real effort to find a solution that fit the type of input data. Furthermore, I think you are on the right track, but you need to polish your code and make it more readable and maintainable.

**Comment by omarmhaimdat** [comment]
I will close this pull request, you will receive an answer by email on Monday.
